include/oneapi/dpl/pstl/utils_ranges.h

```diff
@@ -27,6 +27,35 @@ namespace oneapi
 {
 namespace dpl
 {
+
+namespace __internal
+{
+
+template <typename _R>
+auto
+get_value_type(int) -> typename ::std::decay_t<_R>::value_type;
+
+template <typename _R>
+auto
+get_value_type(long) ->
+    typename ::std::iterator_traits<::std::decay_t<decltype(::std::declval<_R&>().begin())>>::value_type;
+
+template <typename _R>
+auto
+get_value_type(...)
+{
+    //static_assert has to be evaluated to false always - it should depends on a template parameter
+    static_assert(std::is_same_v<_R, void>, "error: 'value_type' is not specified");
```

Comment:
I believe it should be better that that. I mean it doesn't tell the user the root cause of the error. I assume the root cause is the range or view doesn't satisfy the requirements we expect from the passed type. It should be formulated and stated for better compile-time diagnostics.

Reply:
I think `"error: 'value_type' is not specified" (for given type)` pretty clear.
Explaining some C++ tricks like  `std::is_same_v<_R, void>` is redundant.